scrapy/utils/misc.py

```diff
@@ -33,10 +33,19 @@ def arg_to_iter(arg):
 def load_object(path):
     """Load an object given its absolute object path, and return it.
 
-    object can be a class, function, variable or an instance.
+    If ``path`` is not a string, but an object, then return it as is.
+
+    The object can be a class, function, variable or an instance.
     path ie: 'scrapy.downloadermiddlewares.redirect.RedirectMiddleware'
     """
 
+    if not isinstance(path, six.string_types):
+        if callable(path):
```

Comment:
What if it’s a non-callable variable or object?

Reply:
Sorry this has languished so.

But I'm unsure how to treat the logic then.
`callable()` will return True even for some objects that aren't (like an empty class).
But for variables, it's not as simple, if they're strings they could be meant as a classpath to load, or a variable to pass as-is. The distinction would be hard to make.

So should the logic for `load_object` just return anything that doesn't happen to be a string which has a dot (the check for valid classpath), or fails to import?

Reply:
I’m OK with checking that is it not a string and returning it otherwise; I assume a setting that is loaded using `load_object` will not allow a string as value, and we really not want to silence import errors due to typos.

Reply:
https://github.com/scrapy/scrapy/pull/3873#pullrequestreview-471665059